### DISCUSSION TITLE
It seems that DataGridCheckBoxColumn in Datagrid does not respond to clicks directly

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14874] [Q&A]
When the row where DataGridCheckBoxColumn is located is not selected, clicking on checkbox for the first time will not change the status of the checkbox. Instead, it will select the entire row. Only when clicking on the checkbox after selecting the row, will it respond normally. I don't know if this is how it was originally designed.

**Comment by timunie** [comment]
It is by-design, A cell needs to be in edit mode before updating any values. Use `DataGridTemplateColumn` and add a CheckBox into the CellTemplate should unblock you.
